### PR TITLE
[java] Fix #4503: A false negative about JUnitTestsShouldIncludeAssert and testng

### LINKED ISSUE #4503: [java] JUnitTestsShouldIncludeAssert: false negative with TestNG
**Affects PMD Version:7.0.0-rc1** 

**Rule: JUnitTestsShouldIncludeAssert**

Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#junittestsshouldincludeassert>

**Description:**
The first code sample uses the annotations `org.testng.annotations.*`, and the second one uses `org.junit.*`. The analysis results of the two cases should be equivalent because these annotations are equivalent. However, PMD reports a warning in the second sample and no warnings in the first sample. PMD should report a warning in `Case 1` like `Case 2`.

**Code Sample demonstrating the issue:**
**Case 1**
```java
import org.testng.annotations.Test;
import org.testng.annotations.AfterClass;
public class A {
  @AfterClass
  @Test
  void test() {}  // report no warning
}
```
**Case 2**
```java
import org.junit.jupiter.api.Test;
import org.junit.AfterClass;
public class B {
    @AfterClass
    @Test
    void test() {}  // report a warning
}
```

**Expected outcome: The analysis results of the two classes should be the same and both should be reported a violation.**

PMD reports no violation in the first class, but that's wrong. That's a false negative. PMD should treat `testng` like `junit`.

**Running PMD through:** *CLI*

### PR DESCRIPTION
**Pull request by LynnBroe**
## Describe the PR

This PR uses method `isTestMethod` instead method `isJUnitMethod`.

## Related issues

- Fixes #4503 

## Ready?

- [x] Added unit tests for fixed bug/feature
- [x] Passing all unit tests
- [x] Complete build `./mvnw clean verify` passes (checked automatically by github actions)
- [ ] Added (in-code) documentation (if needed)

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to master:<br />
This changeset changes 0 violations,<br />
introduces 10 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 13 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/1a7d08d10fc24819991b6918c7c47eef/diff1/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by oowekyala** [comment]
I suppose we should rename the rule `TestWithoutAssert` or so, since it is not JUnit specific anymore.

**Comment by jsotuyod** [comment]
Yeah, that would apply to several JUnit rules actually. And there are other misnomers:

* `JUnit4TestShouldUseBeforeAnnotation` actually checks for a correct transition from JUnit 3 to either 4 or 5.
* Same applies to `JUnit4TestShouldUseTestAnnotation`
* `JUnitAssertionsShouldIncludeMessage` applies to both, `JUnit` and `TestNg`
* `JUnitTestContainsTooManyAsserts` applies to both, `JUnit` and `TestNg`
* With this PR, `JUnitTestsShouldIncludeAssert` applies to both, `JUnit` and `TestNg`

I'll create a separate issue from this comment to track these required renames / doc changes.

File: pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/bestpractices/xml/JUnitTestsShouldIncludeAssert.xml [hunk 1]
```diff
@@ -705,7 +705,22 @@ public class BigTest {
 }
         ]]></code>
     </test-code>
-
+    
+    <test-code>
+        <description> [java]A false negative about JUnitTestsShouldIncludeAssert and testng #4503 </description>
```
Comment (adangel) [inline pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/bestpractices/xml/JUnitTestsShouldIncludeAssert.xml:710]:
```suggestion
        <description>[java] JUnitTestsShouldIncludeAssert: false negative with TestNG #4503</description>
```

**Review by adangel — APPROVED** [review]
Thanks!

**Pull request merged**